geoips/interfaces/base.py

```diff
@@ -920,3 +969,88 @@ def test_interface(self):
                 output["family"][curr_id] = curr_family
                 output["docstring"][curr_id] = output["func"][curr_id].docstring
         return output
+
+
+class BaseTextInterface(ABC, BaseInterface):
+    """Base Class for GeoIPS Text-Based Interfaces.
+
+    This class should not be instantiated directly. Instead, interfaces should be
+    accessed by importing them from ``geoips.interfaces``. For example:
+    ```
+    from geoips.interfaces import algorithms
+    ```
+    will retrieve an instance of ``AlgorithmsInterface`` which will provide access to
+    the GeoIPS algorithm plugins.
+
+    Inherits from BaseInterface Class, as all other Base<Type>Interface classes do.
+    """
+
+    @property
+    @abstractmethod
+    def plugin_class(self):
+        """The corresponding class that will be used to create the text-plugin."""
+        pass
+
+    def __init__(self):
+        """Initialize module plugin interface."""
+        try:
+            self.text_registry = self.plugin_registry.registered_plugins["text_based"]
+        except KeyError:
+            # Plugin Registry hasn't been created yet, just pass for now.
+            pass
```

Comment:
This should become a property `BaseInterface.interface_registry` and rename `BaseInterface.plugin_registry` to simply `BaseInterface.registry`.

The property can just look like this I think (with proper error handling), no need for caching:
```
@property
def interface_registry(self):
    return self.plugin_registry.registered_plugins[self.interface_type][self.name]
```

This will require adding `self.interface_type` as an abstract property, then add it to all subclasses.

Reply:
Done!